### DISCUSSION TITLE
Emergency plan

### DISCUSSION THREAD

**Discussion by vitobotta** [dani-garcia/vaultwarden#4103] [Q&A]
Hi, I just switched from managed Bitwarden to self hosted Vaultwarden. Thanks for this project!

What is the common recovery strategy with Vaultwarden in case you lose access to both the server and your devices? I was thinking to periodically store a dump of Vaultwarden's postgres database in my email, since I always remember the IMAP credentials. Would this be safe enough? I mean, if I understand it correctly all the data in the postgres database is encrypted, right?

If this is not a good idea, how can I ensure I can always recover access to a copy of my vault if I lose access to my server and devices? Is there a way?

Thanks!

**Comment by GeekCornerGH** [comment]
Database is encrypted indeed. However, you'll want to save the whole /data folder with the backup, as it contains important informations such as server keys for example. Unless you want a backup that you can't use anymore. Also, it would be a better idea to store it on a few usb keys you keep offline most of the time, and at secure places.

**Reply by vitobotta** [reply]
Is the data in the data folder safe to store in emails as well as the database? Thanks!

**Reply by GeekCornerGH** [reply]
Wouldn't recommend in emails, instead on some actual usb sticks you put somewhere in your home and somewhere else (like your friend house) otherwise, you may if you zip it properly

**Reply by vitobotta** [reply]
The scenario I am trying to cover is if I lose access to both my server (problem with provider) and all my devices due to a fire or something in my home. Where I live I don't have anyone nearby who could keep a copy for me and anyway it would be a hassle to keep it up to date. What if I encrypt the archive including the database and the other files with openssl and a strong password I can remember? I guess that would be fine?

**Comment by zvrh** [comment]
https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault
The database contains at least the plaintext main account email address and 2FA code. You can use openssl to re-encrypt the backup.
